### DISCUSSION TITLE
MonaiLabel Models; Training from Scratch

### DISCUSSION THREAD

**Discussion by jesamir** [Project-MONAI/MONAILabel#1714] [Q&A] [answered]
Hi everyone! I am a novice at 3D Slicer, but I am super excited to get started!

I have already completed the tutorials MonaiLabel offers with the Radiology App and the MONAIBundle app which are both consistent with 3D Slicer. However, I am struggling to implement a DeepEdit model. I have attached a picture of the error I keep receiving regardless of the dataset.

![runtimeerror](https://github.com/Project-MONAI/MONAILabel/assets/171720856/4ae74c6d-7cf7-4410-90af-e1549079799c)

I went into the config file of DeepEdit through text editor and changed it into single label and put the background as 0, pancreas as 1, and also put the use_pretrained_model as false. I am trying to attempt segmenting a bunch of different anatomical parts soon with the largest goal to be understanding Active Learning. Reading through the README.md file of the radiology app has made me realize that the DeepEdit model is the most useful for Active Learning strategies while the segmentation model is basic for automated segmentation (correct me if I’m wrong please!). I have watched the Youtube video by MonaiLabel on Training from Scratch which they use the segmentation model. However, if I do want to learn the most about Active Learning, what can I do to achieve this? Or if there is a tutorial someone can recommend me as I have been using the Training from Scratch video and the README.md files to steer me in the right direction. I would really appreciate any help!!

**Comment by diazandr3s** [comment]
Hi @jesamir,

From the error, it seems you're trying to use a pretrained model with a different number of segments. I'd recommend you delete all files within the radiology/model folder.

Also, to make use of Active Learning strategies, there should be a pretrained model on the task you wish the model to work.

**Reply by jesamir** [reply]
Thank you so much! If I do want to train a model from scratch instead of a pretrained model, would Active Learning strategies not become relevant?

**Reply by jesamir** [reply]
I am trying to utilize the DeepEdit model with just 2 labels: lung and background. That is when I receive the RuntimeError. I tried it again with a new radiology app and new data. It still gives me the same error. I tried to just leave the labels and manually change them for every image, but that gets computationally expensive. Thank you in advance for your help, and I look forward to your response!

**Reply by diazandr3s** [reply, marked as answer]
Hi @jesamir,

If you change the number of labels, this is considered a new task. So training from scratch is recommended.

Uncertainty-based active learning strategies need a pretrained model. If you'd like to train a model from scratch on a different number of labels, you'd need to annotate a small number of volumes to kick-off the active learning loop.


Hope this helps,

**Reply by jesamir** [reply]
Awesome! Thank you so much for your help!
